### DISCUSSION TITLE
How do you add a light?

### DISCUSSION THREAD

**Discussion by JohnGilbertson** [sceneview/sceneview#87] [Q&A]
I used sceneform ages ago and could add a light that can be moved and attached to the camera for example with something like:

        val spotlightBuilder= Light.builder(Light.Type.SPOTLIGHT)
        spotlightBuilder.setFalloffRadius(20f)
        spotlightBuilder.setOuterConeAngle(0.2f).setInnerConeAngle(0.1f)
        spotlightBuilder.setIntensity(200f)
        val light=spotlightBuilder.build()
        val spotNode= Node()
        spotNode.light=light

However there doesn't seem to be the ability to add a light to a Node anymore. How do I add the light to the scene?

I've been through trying to use EntityManager, lightManager etc, but even though the code compiles and throws no errors, no usage of that seems to actually affect any lighting in the scene.

**Comment by ThomasGorisse** [comment]
Did you try using [LightNode](https://github.com/SceneView/sceneview-android/blob/main/sceneview/src/main/java/io/github/sceneview/node/LightNode.kt) directly?
I have to admit that we didn't worked/tested this feature a lot since it's rarely used.

You can also handle it manually (Filament Builder without nodes) and add it to the `sceneView.scene`.
Please let us know the issues you could encounter with `LightNode` and have a look at how the default main light is handled within the `SceneView`:
https://github.com/SceneView/sceneview-android/blob/2c8b61de13e6c1a36a7f33fb805a5fba23f45e51/sceneview/src/main/java/io/github/sceneview/SceneView.kt#L216-L222

**Reply by JohnGilbertson** [reply]
Thankyou very much. Somehow I missed LightNode in all of the auto complete options.

**Comment by radcli14** [comment]
Can you help with how to initialize the `LightNode`? I tried the code below, but the `lightInstance` is not the correct form for the initializer. Also tried setting `sceneView.mainLight = light` but that had seemingly no effect.

Thank you.

```
val light = LightManager.Builder(LightManager.Type.POINT).apply {
    val (r, g, b) = Colors.cct(6_500.0f)
    color(r, g, b)
    intensity(100_000.0f)
    castShadows(true)
    position(0f, 1f, 2f)
}
.build(lifecycle)
val lightInstance = light.instance
val lightNode = LightNode(lightInstance)
sceneView.addChild(lightNode)
```
